Sources/Sentry/SentrySessionReplayIntegration.m

```diff
@@ -359,6 +367,43 @@ - (void)moveCurrentReplay
     }
 }
 
+- (void)cleanUp
+{
+    NSURL *replayDir = [self replayDirectory];
+    NSDictionary<NSString *, id> *lastReplayInfo = [self lastReplayInfo];
+    NSString *lastReplayFolder = lastReplayInfo[@"path"];
+
+    // Mapping replay folder here and not in dispatched queue to prevent a race condition between
+    // listing files and creating a new replay session.
```

Comment:
The dispatch queue in the dependency manager is a serial queue, so there shouldn't be a danger of race condition: https://github.com/getsentry/sentry-cocoa/blob/2d2068d4247d91a9143c43c4681805caa06d0443/Sources/Sentry/SentryDispatchQueueWrapper.m#L13-L14

In fact, you might even be creating a race condition by not placing this on the serial queue, depending on what other replay operations are or aren't also performed on the queue. Something that was previously deferred might be interleaved with this line and the next block enqueued below in various combinations. 

For instance, what happens if multiple calls to `SentryReplayApi.start` are received in quick succession?

Reply:
> The dispatch queue in the dependency manager is a serial queue

But the `SentrySessionReplay` class that creates a new replay session has it own queue.

> what happens if multiple calls to SentryReplayApi.start

The Integration is initialized only once and the cleanUp will run only once.

Reply:
Hm, the only reason I mentioned the dependency manager's queue is because that's what's being used in this file. Should we use one queue for all replay activity? Why the dependency manager's here and the private queue in SentrySessionReplay.swift, wouldn't that open up the possibility of more race conditions?

Reply:
No, I dont think there is a need for shared dispatch queue, the integration do some setup, and SessionReplay do the rest.